node/modules/lp2p/pubsub.go

```diff
@@ -31,8 +43,132 @@ func PubsubTracer() PubsubOpt {
+
+					// topic parameters
+					Topics: map[string]*pubsub.TopicScoreParams{
+						build.BlocksTopic(nn): {
+							// expected 10 blocks/min
+							TopicWeight: 0.1, // max is 250, max mesh penalty is -10, single invalid message is -100
+
+							// 1 tick per second, maxes at 1 after 1 hour
+							TimeInMeshWeight:  0.00027, // ~1/3600
+							TimeInMeshQuantum: time.Second,
+							TimeInMeshCap:     1,
+
+							// deliveries decay after 1 hour, cap at 100 blocks
+							FirstMessageDeliveriesWeight: 25,      // max value is 2500
+							FirstMessageDeliveriesDecay:  0.99972, // 1 hour
+							FirstMessageDeliveriesCap:    100,     // 100 blocks
+
+							// deliveries decay after 1 hour, penalty activates at 1 minute and expects ~0.4 blocks
+							MeshMessageDeliveriesWeight:     -576,    // max penalty is -100
+							MeshMessageDeliveriesDecay:      0.99972, // 1 hour
+							MeshMessageDeliveriesCap:        100,     // 100 blocks
+							MeshMessageDeliveriesThreshold:  0.41666, // 5 / 12 blocks/min
+							MeshMessageDeliveriesWindow:     10 * time.Millisecond,
+							MeshMessageDeliveriesActivation: time.Minute,
+
+							// decays after 15 min
+							MeshFailurePenaltyWeight: -576,
+							MeshFailurePenaltyDecay:  0.99888,
+
+							// invalid messages decay after 1 hour
+							InvalidMessageDeliveriesWeight: -1000,
+							InvalidMessageDeliveriesDecay:  0.99972,
+						},
+						build.MessagesTopic(nn): {
+							// expected > 1 tx/second
+							TopicWeight: 0.05, // max is 50, max mesh penalty is -5, single invalid message is -50
+
+							// 1 tick per second, maxes at 1 hour
+							TimeInMeshWeight:  0.0002778, // ~1/3600
+							TimeInMeshQuantum: time.Second,
+							TimeInMeshCap:     1,
+
+							// deliveries decay after 10min, cap at 1000 tx
+							FirstMessageDeliveriesWeight: 1, // max value is 1000
+							FirstMessageDeliveriesDecay:  0.99833,
+							FirstMessageDeliveriesCap:    1000,
+
+							// deliveries decay after 10min, penalty activates at 1 min and expects 5 txs
+							MeshMessageDeliveriesWeight:     -4, // max penalty is -100
+							MeshMessageDeliveriesDecay:      0.99833,
+							MeshMessageDeliveriesCap:        1000,
+							MeshMessageDeliveriesThreshold:  5,
+							MeshMessageDeliveriesWindow:     10 * time.Millisecond,
+							MeshMessageDeliveriesActivation: time.Minute,
+
+							// decays after 5min
+							MeshFailurePenaltyWeight: -4,
+							MeshFailurePenaltyDecay:  0.99666,
+
+							// invalid messages decay after 1 hour
+							InvalidMessageDeliveriesWeight: -1000,
+							InvalidMessageDeliveriesDecay:  0.99972,
+						},
+					},
+				},
+				&pubsub.PeerScoreThresholds{
+					GossipThreshold:             -500,
+					PublishThreshold:            -1000,
+					GraylistThreshold:           -2500,
+					AcceptPXThreshold:           1000,
+					OpportunisticGraftThreshold: 2.5,
+				},
+			),
+		}
+
+		// enable Peer eXchange on bootstrappers
+		if isBootstrapNode {
```

Comment:
we can probably add an `--enable-peer-exchange` flag that passes an option through instead of having to rely on peer ID matching

Reply:
yeah, that would work.

Reply:
it might be better to have a specific option for bootstrappers, let's call it `--bootstrapper`, because there are potentially more tweaks we can do.
One of them would be to run them in gossip+px only mode, where the overlay `D` parameters are set to 0.